dev/Scroller/SnapPointWrapper.cpp

```diff
@@ -36,72 +36,168 @@ int SnapPointWrapper<T>::CombinationCount() const
+        m_actualImpulseApplicableZone);
+    snapPoint->UpdateRestingPointExpressionAnimationForImpulse(
+        m_restingValueExpressionAnimation,
+        m_ignoredValue,
+        m_actualImpulseApplicableZone);
+
+    *conditionalExpressionAnimation = m_conditionExpressionAnimation;
+    *restingPointExpressionAnimation = m_restingValueExpressionAnimation;
+}
+
+// Invoked on pre-RS5 versions when Scroller::m_isInertiaFromImpulse changed
+// and the 'iIFI' boolean parameters need to be updated.
+template<typename T>
+void SnapPointWrapper<T>::GetUpdatedExpressionAnimationsForImpulse(
+    bool isInertiaFromImpulse,
+    winrt::ExpressionAnimation* conditionalExpressionAnimation,
+    winrt::ExpressionAnimation* restingPointExpressionAnimation)
+{
+    MUX_ASSERT(!SharedHelpers::IsRS5OrHigher());
+
+    SnapPointBase* snapPoint = GetSnapPointFromWrapper(this);
+
+    snapPoint->UpdateExpressionAnimationForImpulse(
+        m_conditionExpressionAnimation,
+        isInertiaFromImpulse);
+    snapPoint->UpdateExpressionAnimationForImpulse(
+        m_restingValueExpressionAnimation,
+        isInertiaFromImpulse);
+
+    *conditionalExpressionAnimation = m_conditionExpressionAnimation;
+    *restingPointExpressionAnimation = m_restingValueExpressionAnimation;
 }
 
 template<typename T>
 void SnapPointWrapper<T>::DetermineActualApplicableZone(
     SnapPointWrapper<T>* previousSnapPointWrapper,
-    SnapPointWrapper<T>* nextSnapPointWrapper)
+    SnapPointWrapper<T>* nextSnapPointWrapper,
+    bool forImpulseOnly)
 {
-    winrt::SnapPointBase winrtSnapPoint = safe_cast<winrt::SnapPointBase>(m_snapPoint);
-    SnapPointBase* snapPoint = winrt::get_self<SnapPointBase>(winrtSnapPoint);
+    SnapPointBase* snapPoint = GetSnapPointFromWrapper(this);
     SnapPointBase* previousSnapPoint = GetSnapPointFromWrapper(previousSnapPointWrapper);
     SnapPointBase* nextSnapPoint = GetSnapPointFromWrapper(nextSnapPointWrapper);
+    double previousIgnoredValue = previousSnapPointWrapper ? previousSnapPointWrapper->m_ignoredValue : NAN;
+    double nextIgnoredValue = nextSnapPointWrapper ? nextSnapPointWrapper->m_ignoredValue : NAN;
 
-    m_actualApplicableZone = snapPoint->DetermineActualApplicableZone(
+    if (!forImpulseOnly)
+    {
+        m_actualApplicableZone = snapPoint->DetermineActualApplicableZone(
+            previousSnapPoint,
+            nextSnapPoint);
+    }
+
+    m_actualImpulseApplicableZone = snapPoint->DetermineActualImpulseApplicableZone(
         previousSnapPoint,
-        nextSnapPoint);
+        nextSnapPoint,
+        m_ignoredValue,
+        previousIgnoredValue,
+        nextIgnoredValue);
 }
 
 template<typename T>
 void SnapPointWrapper<T>::Combine(SnapPointWrapper<T>* snapPointWrapper)
 {
-    winrt::SnapPointBase winrtSnapPoint = safe_cast<winrt::SnapPointBase>(m_snapPoint);
-    SnapPointBase* snapPoint = winrt::get_self<SnapPointBase>(winrtSnapPoint);
+    SnapPointBase* snapPoint = GetSnapPointFromWrapper(this);
```

Comment:
I'd pobably just combine these two lines, assuming GetSnapPointFromWrapper doesn't ever return null, which we aren't checking for anyway

Reply:
Done.